File: 백준/Silver/11659. 구간 합 구하기 4/구간 합 구하기 4.cs
using System;
using System.IO;
using System.Text;

namespace AlgorithmStudy
{
    class boj11659
    {
        public static void Main(string[] args)
        {
            StreamReader sr = new StreamReader(new BufferedStream(Console.OpenStandardInput()));
            StringBuilder sb = new StringBuilder();
            string[] input = sr.ReadLine().Split();
            int n = int.Parse(input[0]), m = int.Parse(input[1]);
            int[] arr = new int[n + 1];
            input = sr.ReadLine().Split();
            arr[1] = int.Parse(input[0]);
            for (int i = 2; i <= n; i++)
                arr[i] = arr[i - 1] + int.Parse(input[i - 1]);
            for (int i = 0; i < m; i++)
            {
                input = sr.ReadLine().Split();
                sb.AppendLine((arr[int.Parse(input[1])] - arr[int.Parse(input[0]) - 1]).ToString());
            }
            sr.Close();
            Console.Write(sb);
        }
    }
}

File: 백준/Silver/11659. 구간 합 구하기 4/구간 합 구하기 4.cc
#include <iostream>
#include <string>
using namespace std;

int main() {
	string result;
	int arr[100000] = { 0 };
	int n, m;
	cin >> n >> m;
	for (int i = 0; i < n; i++) {
		cin >> arr[i];
		if (i > 0)
			arr[i] += arr[i - 1];
	}

	for (int i = 0; i < m; i++) {
		int a, b;
		cin >> a >> b;
		a -= 2; b -= 1;
		if (a < 0)
			result.append(to_string(arr[b]) + "\n");
		else
			result.append(to_string(arr[b] - arr[a]) + "\n");
	}

	result.erase(result.end() - 1);
	cout << result;

	return 0;
}